functions/database/database.js

```diff
@@ -0,0 +1,316 @@
+ *
+ * @param {string} releaseId The ID of the release to fetch.
+ * @return {Promise<Object>} A promise that resolves to the release data.
+ */
+async function getReleaseData(releaseId) {
+  const releaseSnapshot = await db.collection("releases")
+      .doc(releaseId)
+      .get();
+
+  if (!releaseSnapshot.exists) {
+    throw new Error(`No release found with ID: ${releaseId}`);
+  }
+
+  const releaseData = releaseSnapshot.data();
+
+  return releaseData;
+}
+
+/** Retrieve all past releases from Firestore. This includes releases that are
+ * currently active.
+ *
+ * @return {Promise<Array>} Returns a promise that resolves to an array
+ * of Firestore document data of past releases.
+ * @throws {Error} If there is an error in fetching data from Firestore,
+ * the promise is rejected with an error.
+ */
+async function getPreviousReleaseData() {
+  const today = new Date();
+  const snapshot = await db.collection("releases")
+      .where("releaseDate", "<", today)
+      .get();
+
+  const pastReleases = snapshot.docs.map((doc) => doc.data());
+
+  return pastReleases;
+}
+
+/** Write new releases to Firestore. This function assumes that the contents
+ * of the releases are validated. newReleases is an array of release objects
+ * that have the following structure:
+ * {
+ *   releaseName: string,
+ *   releaseOperator: string,
+ *   codeFreezeDate: Date,
+ *   releaseDate: Date,
+ * }
+ *
+ * @param {Object} newReleases - Releases to store in Firestore
+ */
+async function addReleases(newReleases) {
+  // Validate the structure of the new releases before we store
+  validateNewReleaseStructure(newReleases);
+
+  const batch = db.batch();
+
+  newReleases.forEach((release) => {
+    const newReleaseRef = db.collection("releases").doc();
+    const snapshotBranchName = `releases/${release.releaseName}`;
+    const releaseBranchName = `releases/${release.releaseName}.release`;
+    const snapshotBranchLink = `${REPO_URL}/tree/${snapshotBranchName}`;
+    const releaseBranchLink = `${REPO_URL}/tree/${releaseBranchName}`;
+    const releaseData = {
+      state: RELEASE_STATES.SCHEDULED,
+      releaseName: release.releaseName,
+      releaseOperator: release.releaseOperator,
+      codeFreezeDate: release.codeFreezeDate,
+      releaseDate: release.releaseDate,
+      snapshotBranchName: snapshotBranchName,
+      snapshotBranchLink: snapshotBranchLink,
+      releaseBranchName: releaseBranchName,
+      releaseBranchLink: releaseBranchLink,
+      isComplete: false,
+      buildArtifactStatus: "",
+      buildArtifactConclusion: "",
+      buildArtifactLink: "",
+      buildArtifactJobId: "",
+    };
+    batch.set(newReleaseRef, releaseData);
+  });
+
+  await batch.commit();
+}
+
+/**
+   * Delete the releases in Firestore with code freeze dates that are in the
+   * future. This is intented to be used when scheduling new upcoming releases,
+   * but we need to delete the old upcoming releases before.
+   */
```

Comment:
nit: 
Separate into title + description:
```js
/**
   * Delete the releases in Firestore with code freeze dates that are in the future. 
   * 
   * This is intented to be used when scheduling new upcoming releases,
   * but we need to delete the old upcoming releases before.
   */
```

Reply:
Resolved in https://github.com/firebase/firebase-release-dashboard/pull/8/commits/26d9dc272eb39d4ca7696fec7a03a8a9d3eb8bab